tests/components/utils-test.js

```diff
@@ -381,6 +381,19 @@ describe('Utils', function () {
 
     var existingVal = Utils.getPropertyValue({ property: { nestedProp: 'value' } }, 'property.nestedProp');
     expect(existingVal).toEqual('value');
+
+    //comaparing the falsy values with === so I'm sure types match and that there is no implicit conversion.
+    var falsyVal = Utils.getPropertyValue({ property: { nestedProp: 0 } }, 'property.nestedProp', 1);
+    expect(falsyVal === 0).toEqual(true);
+
+    falsyVal = Utils.getPropertyValue({ property: { nestedProp: null } }, 'property.nestedProp', 1);
+    expect(falsyVal === null).toEqual(true);
+
+    var stringVal = Utils.getPropertyValue({ property: { nestedProp: { test:'test' } } }, 'property.nestedProp.test', 1);
+    expect(stringVal === 'test').toEqual(true);
+
+    functionVal = Utils.getPropertyValue({ property: { nestedProp: { test:'test' } } }, 'property.nestedProp.toString', 1);
```

Comment:
Should we test a newly defined function in addition to toString, which is already part of an object's prototype? We can also try calling this function returned by Utils.getPropertyValue to see if it works properly.

Reply:
Sure. I'll add it.